src/box/iproto.cc

```diff
@@ -1163,6 +1176,10 @@ iproto_connection_new(struct iproto_thread *iproto_thread, int fd)
 static inline void
 iproto_connection_delete(struct iproto_connection *con)
 {
+	/* con->shutdown_msg may be used in shutdown process. */
+	if (rlist_empty(&con->in_all_list))
```

Comment:
Here is memory leak! In case this `if` is `true`, below code is not executed!

Reply:
`if` is `true` only if shutdown started (else `iproto_thread->all_connections` also in the same list) and after `shutdown_timeout` seconds memory will freed.